src/blockchain/rpc.rs

```diff
@@ -197,132 +207,151 @@ impl Blockchain for RpcBlockchain {
+                            let related_txs = listed_txs
+                                .iter()
+                                .filter(|tx_result| {
+                                    // Filter out txs related to the script_pubkey
+                                    if let Some(address) = &tx_result.detail.address {
+                                        if address.script_pubkey() == *sk {
+                                            // If conftime data is available, cache them.
+                                            if let (Some(height), Some(timestamp)) = (
+                                                tx_result.info.blockheight,
+                                                tx_result.info.blocktime,
+                                            ) {
+                                                txid_conftime_map.insert(
+                                                    tx_result.info.txid,
+                                                    ConfirmationTime { height, timestamp },
+                                                );
+                                            }
+                                            true
+                                        } else {
+                                            // address doesn't match
+                                            false
+                                        }
+                                    } else {
+                                        // There is no address associated with this tx
+                                        false
+                                    }
+                                })
+                                .map(|tx_result| (tx_result.info.txid, tx_result.info.blockheight))
+                                .collect::<Vec<_>>();
+                            related_txs
+                        })
+                        .collect::<Vec<_>>();
+
+                    script_req.satisfy(satisfier)?
                 }
 
-                for input in tx.input.iter() {
-                    if let Some(previous_output) = db.get_previous_output(&input.previous_output)? {
-                        sent += previous_output.value;
-                    }
+                Request::Tx(tx_req) => {
+                    let tx_needed = tx_req.request();
+
+                    let satisfier = tx_needed
+                        .map(|txid| {
+                            let full_tx = self.client.get_transaction(txid, Some(true))?;
+                            let tx = deserialize::<bitcoin::Transaction>(&full_tx.hex)?;
+                            let prev_outs = tx
+                                .input
+                                .iter()
+                                .map(|txin| {
+                                    // check if the prev_out is in db
+                                    if let Some(txout) =
+                                        db.get_previous_output(&txin.previous_output)?
```

Comment:
I think blockchain logic should not do this. Rather when you pass a `None` to `Option<Vec<TxOut>>` when you satisfy the request the TxReq can look it up in the database to see if it can salvage it.

Reply:
Ah that's pretty neat. But what if the prev_out is not in the db? Will that cause any problem? This filter is added here because to satisfy the TxReq, I need to know weather I have to fetch for the prev_out, which is costly, so I don't wanna do that for all the inputs. Thus the filter. 

As @RCasatta mentioned here https://github.com/bitcoindevkit/bdk/pull/466#issuecomment-964068836, if it's possible to partially satisfy the TxReq without all the prev_outs, then the `txindex` requirement problem will also get solved.

Reply:
You're right my thinking was just wrong.